src/routing.js

```diff
@@ -0,0 +1,76 @@
+import {chain, merge, toPairs, trim} from 'lodash';
+
+const isParam = segment => segment.substr(0, 1) === ':';
+const toSegments = pathname => trim(pathname, '/').split('/');
+
+const buildRoute = (segments, methods) => {
+  const segment = segments.shift();
```

Comment:
`shift` mutates the `segments` array, right? I think we should keep these mutation / side effects out of a function.
Try using the [array spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) instead, e.g.

```js
const [nextSegment, ...remainingSegments] = segments;
const isLastSegment = !(remainingSegments && remainingSegments.length > 0);
```